packages/block-library/src/cover/index.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Server-side rendering of the `core/categories` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/categories` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the categories list/dropdown markup.
+ */
+function render_block_core_cover( $attributes, $content ) {
+	if( $attributes['useFeaturedImage'] === false ) {
+		return $content;
+	}
+
+	$currentFeaturedImage = get_the_post_thumbnail_url();
```

Comment:
Given that this block might be used outside of WP should we use a `function_exists` here.

Also is there a way whereby the value could be made extensible by way of a filter that would allow other systems using the block editor to provide their own URL?

Reply:
We don't do that check in other blocks that depend on WP as far as I can see. E.g. `archives` or `latest_coments` blocks call WP functions with no checks. By adding this dependency on the post's featured image the cover block becomes just as WP centric for now. Plus, there is more to the dynamic-ness of the block that should be handled if the block is used elsewhere.

Currently if one disables the `useFeaturedImage` attribute the block juts works.

Reply:
> Also is there a way whereby the value could be made extensible by way of a filter that would allow other systems using the block editor to provide their own URL?

Sure we could add a filter, but why? The featured image itself can be filtered and the block is set to _use the featured image_, so it should respect the intention set by the toggle.

Reply:
I was thinking it would be great to filter the value if you were using the block outside of WP, but then I realised filters _are_ part of WordPress.

Guess this block is coupled to WP.